demos/mqtt/mqtt_demo_serializer/mqtt_demo_serializer.c

```diff
@@ -851,34 +878,67 @@ int main( int argc,
+
+            do
+            {
+                /* The client is now connected to the broker. Subscribe to the topic
+                 * as specified in MQTT_EXAMPLE_TOPIC at the top of this file by sending a
+                 * subscribe packet then waiting for a subscribe acknowledgment (SUBACK).
+                 * This client will then publish to the same topic it subscribed to, so it
+                 * will expect all the messages it sends to the broker to be sent back to it
+                 * from the broker. This demo uses QOS0 in subscribe, therefore, the Publish
+                 * messages received from the broker will have QOS0. */
+                /* Subscribe and SUBACK */
+                LogInfo( ( "Attempt to subscribe to the MQTT topic %s\r\n", MQTT_EXAMPLE_TOPIC ) );
+                mqttSubscribeToTopic( &networkContext, &fixedBuffer );
+
+                /* Since subscribe is a control packet, record the last control packet sent
+                 * timestamp. This timestamp will be used to determine if it is necessary to
+                 * send a PINGREQ packet. */
+                returnStatus = clock_gettime( CLOCK_MONOTONIC, &currentTimeStamp );
+                assert( returnStatus == 0 );
+                lastControlPacketSentTimeStamp = currentTimeStamp.tv_sec;
+
+                /* Process incoming packet from the broker. After sending the subscribe, the
+                 * client may receive a publish before it receives a subscribe ack. Therefore,
+                 * call generic incoming packet processing function. Since this demo is
+                 * subscribing to the topic to which no one is publishing, probability of
+                 * receiving Publish message before subscribe ack is zero; but application
+                 * must be ready to receive any packet.  This demo uses the generic packet
+                 * processing function everywhere to highlight this fact. */
+                mqttProcessIncomingPacket( &networkContext, &fixedBuffer );
+
+                /* Check status of suback sent from broker. If server rejected the subscription
+                 * request, attempt resubscription to the topic filter. */
+                if( globalSubAckStatus == false )
+                {
+                    /* Send PINGREQ to the broker. A PINGREQ is sent to avoid hitting keep-alive
+                     * time-out period during backoff and sleep execution, before the next
+                     * subscription attempt. */
+                    LogInfo( ( "Sending PINGREQ to the broker\n " ) );
+                    mqttKeepAlive( &networkContext, &fixedBuffer );
+
+                    returnStatus = clock_gettime( CLOCK_MONOTONIC, &currentTimeStamp );
+                    assert( returnStatus == 0 );
+                    lastControlPacketSentTimeStamp = currentTimeStamp.tv_sec;
+
+                    mqttProcessIncomingPacket( &networkContext, &fixedBuffer );
```

Comment:
This should have a comment about processing the PING response packet from the broker